Source/Charts/Charts/ChartViewBase.swift

```diff
@@ -905,17 +857,11 @@ open class ChartViewBase: NSUIView, ChartDataProvider, AnimatorDelegate
         }
         set
         {
-            var val = newValue
-            if val < 0.0
-            {
-                val = 0.0
-            }
-            if val >= 1.0
-            {
-                val = 0.999
+            switch newValue {
+            case ..<0.0: _dragDecelerationFrictionCoef = 0
```

Comment:
where can I find this magic definition.. I just can find things like `case 1..<5:` in the manual.

Reply:
[One Sided Ranges(https://github.com/apple/swift-evolution/blob/master/proposals/0172-one-sided-ranges.md) are new in Swift 4.